asset/asset.go

```diff
@@ -1118,12 +1109,9 @@ func (s *ScriptKey) IsEqual(otherScriptKey *ScriptKey) bool {
 // the local wallet or was explicitly declared to be known by using the
 // DeclareScriptKey RPC. Knowing the key conceptually means the key belongs to
 // the local wallet or is at least known by a software that operates on the
-// local wallet. The DeclaredAsKnown flag is never serialized in proofs, so this
-// is never explicitly set for keys foreign to the local wallet. Therefore, if
-// this method returns true for a script key, it means the asset with the script
-// key will be shown in the wallet balance.
+// local wallet.
 func (s *ScriptKey) DeclaredAsKnown() bool {
-	return s.TweakedScriptKey != nil && s.TweakedScriptKey.DeclaredKnown
+	return s.TweakedScriptKey != nil && s.Type != ScriptKeyUnknown
```

Comment:
Maybe `ScriptKeyUnknown` should be renamed to `ScriptKeyTypeUnknown`, because the type of script key is the thing that is unknown, even for a tracked script key. Have i understood that correctly?

Reply:
Yeah, I guess so. But then as a consequence we should also put the word `Type` into the values of the pseudo-enum, so `ScriptKeyTypeBip86`, `ScriptKeyTypeScriptPathExternal` and so on, which makes them even longer.
But since we have an explicit Golang type that also has the word `Type` in it, that also provides some context to it. So as long as we don't have any naming collision I'm opting to keep the current naming.